### PR TITLE
✅ Allowing service-worker-scope attribute in amp-web-push for AMP validator

### PR DESCRIPTION
**Pull request by demianrenzulli**
Adding validation for `service-worker-scope` attribute in `amp-web-push` component (complementing change made in #21702).

The rule created in validator-amp-web-push.protoascii defines the attribute as:

- Optional: By not adding a "mandatory" attribute in the declaration.
- Of type URL, HTTPS and relative: This comes from [SW Spec](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/register#Parameters): 

"**scope**: A USVString representing a URL that defines a service worker's registration scope; that is, what range of URLs a service worker can control. This is usually a relative URL. It is relative to the base URL of the application."

Tests have been added to show the different use cases: 
- Using the component without service-worker-scope.
- Using the component with service-worker-scope, in the root, and a subpath.

@jridgewell: if you can double-check all this is correct, would be great.

Thanks for the help.

### CODE REVIEW

File: extensions/amp-web-push/validator-amp-web-push.protoascii [hunk 1]
```diff
@@ -57,6 +57,13 @@ tags: {  # <amp-web-push>
       allow_relative: false
     }
   }
+  attrs: {
+    name: "service-worker-scope"
+    value_url: {
+      protocol: "https"
```
Comment (jridgewell) [inline extensions/amp-web-push/validator-amp-web-push.protoascii]: Can a scope be an absolute URL?

Comment (demianrenzulli) [reply extensions/amp-web-push/validator-amp-web-push.protoascii]: The [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/register#Parameters) say: "This is **usually** a relative URL", but doesn't say **it has to be** a relative URL. 
That's why I assume the answer is yes.

I also wrote [this test](https://amp-web-push-changes.glitch.me/subpath/index.html), which uses an absolute URL for the "service-worker-scope" attribute, and it looks like it's working well when inspecting the installed SW in CDT "Application" tab.

If you want to double-check, here's [the code of the Glitch test](https://glitch.com/~amp-web-push-changes).

**Comment by demianrenzulli** [comment]
Just added a minor modification to the test:

It was using service-worker-scope="/subpath", and it should be: service-worker-scope="/subpath/".
Both are "valid", but only the second actually registers the SW.

File: extensions/amp-web-push/validator-amp-web-push.protoascii [hunk 2]
```diff
@@ -57,6 +57,13 @@ tags: {  # <amp-web-push>
...
+      allow_relative: true
```
Comment (honeybadgerdontcare) [inline extensions/amp-web-push/validator-amp-web-push.protoascii]: We will need to update AMP caches to rewrite this attribute from relative to absolute before merging. Filed b/131437166. We could move forward now without relative URLs in this PR and allow relative URLs after that issue is addressed.

Comment (jridgewell) [reply extensions/amp-web-push/validator-amp-web-push.protoascii]: Context here, this value is only used when the AMP doc is loaded on origin. It won't be used if served from a cache.

Given that, is the rewrite necessary?

Comment (honeybadgerdontcare) [reply extensions/amp-web-push/validator-amp-web-push.protoascii]: If the Runtime is guaranteeing that then the rewrite is not necessary.

Comment (jridgewell) [reply extensions/amp-web-push/validator-amp-web-push.protoascii]: This is a security guarantee by by the browser, you can't register a service worker unless the origin of the SW and the page's origin match. So there's never really a point to use this unless we're on the original page's origin, meaning it'll be the correct relative URL.

Comment (honeybadgerdontcare) [reply extensions/amp-web-push/validator-amp-web-push.protoascii]: That makes sense, thank you for explaining.

BTW, `true` is the default value for `allow_relative` so the line is unnecessary.

**Comment by demianrenzulli** [comment]
cc // @guilef

**Pull request merged**